packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -692,6 +758,28 @@ export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGr
     }
   }
 
+  private renderMetadataOptions(props: LaunchTemplateProps) {
+    if (props.httpEndpoint == undefined &&
+      props.httpProtocolIpv6 == undefined &&
+      props.httpPutResponseHopLimit == undefined &&
+      props.httpTokens == undefined &&
+      props.instanceMetadataTags == undefined
+    ) {
+      return undefined;
+    } else {
+      return {
+        httpEndpoint: props.httpEndpoint == true ? 'enabled' :
```

Comment:
```suggestion
        httpEndpoint: props.httpEndpoint === true ? 'enabled' :
```

For all of these checks.